### ISSUE TITLE
Estimate delay distributions from data

### ISSUE THREAD

**Issue by thibautjombart** [epiverse-trace/epiparameter#8]
With this feature the package would estimate a delay distribution from the data. The procedure is not entirely trivial as most distributions will need to be discretized: fitting to a continuous distribution and discretising it may be a good first approximation, but no there are no guarantees that this will produce the same estimate as fitting the discrete version directly. 

Some related work:

* [epitrix::fit_disc_gamma](https://github.com/reconhub/epitrix/blob/master/R/fit_discrete.R), which implements the procedure but only for a discretized Gamma distribution
* [fitdistrplus](https://cran.r-project.org/web/packages/fitdistrplus/index.html), a reference package for fitting distributions; it does not seem to handle discretization though
* the [distcrete](https://github.com/reconhub/distcrete) package, which discretizes any distribution

Some nice additional feature would be to be build wrappers for *linelist* objects.

**Comment by sbfnk** [comment]
Also (for discretised lognormal)
- [EpiNow2::estimate_delay](https://epiforecasts.io/EpiNow2/reference/estimate_delay.html)

One issue is that all of these only work with completed delays and thus are potentially subject to [epidemic phase bias](https://doi.org/10.1177/09622802221107105). If wanting to correct for this and potential right truncation one would need an estimate that accounts for it, e.g. what is implemented for a discretised lognormal in
- [EpiNow2::estimate_truncation](https://epiforecasts.io/EpiNow2/reference/estimate_truncation.html)

**Comment by seabbs** [comment]
Noting the existence of primarycensored which extends fitdistrplus to allow fitting of double censored and truncated delay data: https://primarycensored.epinowcast.org/articles/fitting-dists-with-fitdistrplus.html

I imagine some docs would close this issue out pointing at the various options (though I think perhaps would want to limit to linking out to resources that are "correct".

**Comment by seabbs** [comment]
Another option here is you could wrap that support so that it works natively with you distribution objects.

**Comment by adamkucharski** [comment]
I think outsourcing to more specialist packages like primarycensored and epidist (and ensuring compatability with reuse/comparison functions, e.g. if want to pull in some distributions from epireview and see how new real-time estimates commpare) is sensible. Could also create some 'how to' examples showing how to link together various functions (which will hopefully be hosted on collaboratory pages in future) – maybe with H5N1 serial interval as starting point?

**Comment by seabbs** [comment]
yup agree. I think it might be good to think about the interface when doing that as perhaps it will be less nice than it might be for people. If that is the case I think it would suggest having a converter here (especially if #444 adds `primaraycensored` as a dependency anyway). It could be done as s3 (which would require some changes to the `fitdistrplus` support in `primarycensored` I am happy to make and only use `fitdistrplus` as a suggest.

**Comment by seabbs** [comment]
yup agree. I think it might be good to think about the interface when doing that as perhaps it will be less nice than it might be for people. If that is the case I think it would suggest having a converter here (especially if #444 adds `primaraycensored` as a dependency anyway). It could be done as s3 (which would require some changes to the `fitdistrplus` support in `primarycensored` I am happy to make and only use `fitdistrplus` as a suggest.

**Comment by joshwlambert** [comment]
Closing — the direction agreed in this thread is that {epiparameter} should not estimate parameters from data, and that work is better outsourced to specialist packages.

As raised above, naive fitting to completed delays is subject to epidemic phase bias and right truncation, and handling those properly is a substantial undertaking that is already well served elsewhere: {primarycensored}, {epidist}, and {EpiNow2}.

Rather than reimplement any of this, the package now signposts it in a "Related projects" section has been added to the `README` listing the packages above (alongside the sources of already-estimated parameters), and the Scope section of the design principles vignette states explicitly that the package does not estimate parameters from data.

On the converter suggested above remains a good idea and will be taken forward alongside #444, since that issue is where the question of adding {primarycensored} as a dependency is being decided.
